### DISCUSSION TITLE
DOCKER: Unable to start embedded Jetty web server

### DISCUSSION THREAD

**Discussion by Nekima** [alfio-event/alf.io#1041] [Q&A]
Hello,

When I run alf.io with Docker, just by launching the image, with the port mapped correctly, it shows me this error:

>> powered by Spring Boot

20:01:41.953 [main] INFO  alfio.config.SpringBootLauncher - Starting SpringBootLauncher on 7a31073303b2 with PID 1 (/home/alfio/app/WEB-INF/classes started by alfio in /home/alfio/app)

20:01:41.960 [main] INFO  alfio.config.SpringBootLauncher - The following profiles are active: spring-boot,stdout

20:01:43.701 [main] WARN  org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Jetty web server

20:01:43.717 [main] ERROR org.springframework.boot.SpringApplication - Application run failed

org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Jetty web server

You can find the complete log in attachements

Thanks you!

**Comment by cbellone** [comment]
Hi, can you please attach the logs?
thanks

**Reply by Nekima** [reply]
There it is, thanks you!

**Comment by Nekima** [comment]
[alfio.log](https://github.com/alfio-event/alf.io/files/7817677/alfio.log)

**Comment by cbellone** [comment]
you should set the following properties in order to connect to your database:

```
docker [...]
-e POSTGRES_PORT_5432_TCP_PORT=5432 \
-e POSTGRES_PORT_5432_TCP_ADDR=db-hostname \
-e POSTGRES_ENV_POSTGRES_DB=alfio \
-e POSTGRES_ENV_POSTGRES_USERNAME=alfio \
-e POSTGRES_ENV_POSTGRES_PASSWORD=alfio
```

see [Docker compose example](https://github.com/alfio-event/alf.io/blob/master/docker-compose.yml#L6:L10)
